core/capabilities/registry.go

```diff
@@ -14,16 +14,40 @@ var (
 	ErrCapabilityAlreadyExists = errors.New("capability already exists")
 )
 
+type metadataRegistry interface {
+	GetLocalNode(ctx context.Context) (capabilities.Node, error)
+	ConfigForCapability(ctx context.Context, capabilityID string, donID uint32) (capabilities.CapabilityConfiguration, error)
+}
+
 // Registry is a struct for the registry of capabilities.
 // Registry is safe for concurrent use.
 type Registry struct {
-	lggr logger.Logger
-	m    map[string]capabilities.BaseCapability
-	mu   sync.RWMutex
+	metadataRegistry metadataRegistry
+	lggr             logger.Logger
+	m                map[string]capabilities.BaseCapability
+	mu               sync.RWMutex
+}
+
+func (r *Registry) GetLocalNode(ctx context.Context) (capabilities.Node, error) {
+	if r.metadataRegistry == nil {
+		return capabilities.Node{}, errors.New("metadataRegistry information not available")
+	}
+
+	return r.metadataRegistry.GetLocalNode(ctx)
 }
 
-func (r *Registry) GetLocalNode(_ context.Context) (capabilities.Node, error) {
-	return capabilities.Node{}, nil
+func (r *Registry) ConfigForCapability(ctx context.Context, capabilityID string, donID uint32) (capabilities.CapabilityConfiguration, error) {
+	if r.metadataRegistry == nil {
+		return capabilities.CapabilityConfiguration{}, errors.New("metadataRegistry information not available")
+	}
+
+	return r.metadataRegistry.ConfigForCapability(ctx, capabilityID, donID)
+}
+
+func (r *Registry) SetLocalRegistry(lr metadataRegistry) {
```

Comment:
// Sets a local copy of the on-chain registry for the launcher to use.
// Public only for testing purposes.